charts/prometheus/tests/prometheus-config-configmap_test.yaml

```diff
@@ -3,7 +3,31 @@ suite: Test prometheus-config-configmap
 templates:
   - prometheus-config-configmap.yaml
 tests:
-  - it: should work
+  - it: is the correct type of resource
     asserts:
       - isKind:
           of: ConfigMap
+  - it: contains exactly one document
+    asserts:
+      - hasDocuments:
+          count: 1
+  - it: does not include substrings that appear to mix up namespace and helm release name
+    release:
+      name: my-release
+      namespace: my-namespace
+    set:
+      global.blackboxExporterEnabled: true
+      global.veleroEnabled: true
+      global.prometheusPostgresExporterEnabled: true
+      global.nodeExporterEnabled: true
+      tcpProbe.enabled: true
+    asserts:
+      - notMatchRegex:
+          path: data.config
+          pattern: '.*\.my-release\..*'
+      - notMatchRegex:
+          path: data.config
+          pattern: '.*\.my-release:.*'
+      - notMatchRegex:
+          path: data.config
+          pattern: '[^\^]+my-namespace-.*'
```

Comment:
This tests regex matches the prometheus config to error if something like this is detected:

'my-namespace-'
or
'.my-release.'
or
'.my-release:'

typically surrounded by dots or colon means it's supposed to be namespace, and also typically if a dash follows then it should be release name.